### PR TITLE
[Azure Maps] - Onboarding for Search, Route, Render SDKs.

### PR DESCRIPTION
**Pull request by daniel-rocha**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/maps/test-resources.json [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+        "metadata": {
+          "description": "The base resource name."
+        }
+      },
+      "endpointPrefix": {
+        "defaultValue": "communication",
+        "type": "string"
+      },
+      "communicationServicesEndpointSuffix": {
+        "defaultValue": ".communication.azure.com",
```
Comment (alextts627) [inline sdk/maps/test-resources.json:17]: Maybe it's not `communication`?

File: sdk/maps/test-resources.json [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    "parameters": {
+      "baseName": {
+        "type": "string",
+        "defaultValue": "[resourceGroup().name]",
+        "metadata": {
+          "description": "The base resource name."
+        }
+      },
+      "endpointPrefix": {
+        "defaultValue": "communication",
```
Comment (alextts627) [inline sdk/maps/test-resources.json:13]: endpointPrefix is either not needed or change it to out service name `maps`

**Comment by daniel-rocha** [comment]
This parameter  is not used in the rest of the template. It can be deleted.

**Comment by daniel-rocha** [comment]
@alextts627 @alankashiwa @dubiety 

This PR is compromised now and will need to be closed. @khmic5 merged main into maps/onboarding/search-route-render and it caused this PR to include the entire repository in its checks (there are thousands now).

At this point, I can no longer touch the code as I don't have access, so please huddle and fix this yourselves using the following actions:

- I will close this PR. it is not in a good state anymore.
- @khmic5  git clone this repository, git checkout maps/onboarding/search-route-render
- revert the branch so the commit at the tip of the branch is b0f68276b6b96cf9445610fcb0420f6d7d8366a8
- at that point: copy the modified files and folders onto something clean. please note that there are three files that have been changed outside the sdk/maps folder: version_clients.txt, checkstyle-supressions.xml, spotbugs-exclusions.xml. Those also need to be copied out.
- @khmic5 needs to sync her fork with Azure:main, and start clean.
- she should branch from main to a new branch, say maps/khmic5/onboarding
- copy the sdk/maps folders from the previous step onto this new branch, make a commit.
- *do not copy version_client.txt, checkstyle-supressions.xml, spotbugs-exclusions.xml* over their counterparts in their respective folders. Otherwise they will erase many changes by other committers and put *the entire repository in jeopardy*. You don't want that. Rather, see the changes made to those files by me and make the same changes in the files in this branch, so you are starting from the absolutely latest commit from Azure:main.
- once the new onboarding branch (from @khmic5 fork) is ready, with the previous changes copied in and the three files mentioned above correctly manually changed, commit and push.
- create a new DRAFT PR from this new branch.
- run /azp run prepare-pipelines
- verify that there are 77 checks or so. any more or less than this and the pipelines are not correct. this shouldn't happen but if it does then ping me on github.

*Maintaining the repository*

This problem occurred because the main branch was merged into maps/onboarding/search-render-route, which is forbidden. PR branches can only be REBASED. Please @alextts627 @dubiety work with @khmic5 to ensure solid understanding of this practice, to avoid future problems. Practice: syncing the fork against Azure:main (fetch upstream from Github), then from the local computer git pull main, then git checkout maps/khmic5/onboarding, then git rebase main. The problem happened in the last step when the command ran was git merge main. 

Thanks!

**Pull request closed without merging**